### ISSUE TITLE
logger.warning no longer supports UserWarning

### ISSUE THREAD

**Issue by justinessert** [hynek/structlog#477]
## Context

It seems like with the latest version of the `structlog` package, we can no longer pass a `UserWarning` to `logger.warning()`.

## Behavior in 22.1.0 / Expected

Running this would result in the following log `2022-11-22 13:39.48 [warning  ] test`, as expected.

```python
import structlog
logger = structlog.get_logger(__name__)
logger.warning(UserWarning("test"))
```

## Behavior in 22.2.0 / Error

Running the same code will result in the following error.

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/user/miniconda3/envs/env_name/lib/python3.9/site-packages/structlog/_log_levels.py", line 156, in meth
    return self._proxy_to_logger(name, event % args, **kw)
TypeError: unsupported operand type(s) for %: 'UserWarning' and 'tuple'
```

## Workaround

This will replicate the behavior of 22.1, you can wrap the warning in a string, but it would be nice to have this supported as it was.

```python
import structlog
logger = structlog.get_logger(__name__)
logger.warning(str(UserWarning("test")))
```

**Comment by hynek** [comment]
This should be effectively fixed by #375 but I really wonder why you would do that?

**Comment by hynek** [comment]
22.3 is on PyPI
